lib/project/nativescript-project.ts

```diff
@@ -182,6 +184,18 @@ export class NativeScriptProject extends FrameworkProjectBase implements Project
 			}
 		}).future<void>()();
 	}
+
+	public ensureProject(projectDir: string): IFuture<void> {
+		return (() => {
+			if (this.$typeScriptService.isTypeScriptProject(projectDir).wait()) {
+				try {
+					this.$npmService.install(projectDir).wait();
+				} catch (err) {
+					this.$logger.trace(`Failed to install all npm dependencies in the project. Error: ${err}.`);
```

Comment:
I think it is better to display warning here instead of rethrow the exception because if the npm install fails we can continue with the command execution with the current content of the node_modules in the project.